src/main/java/servlets/ResultsServlet.java

```diff
@@ -61,4 +65,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(testRecipes));
   }
+
+  public String[] formatQueryAsList(String query) {
+    query = query.toUpperCase();
+    String[] queryList = query.split(",");
+    for (int idx = 0; idx < queryList.length; idx++) {
+      queryList[idx] = queryList[idx].trim();
+    }
+    return queryList;
+  }
+
+  public Filter generateFiltersFromQuery(String[] queryList) {
+    // Note: Nothing shows up if nothing is put into the search box.
```

Comment:
also, does this work if you remove this special case for 0/1 word queries? I'm curious if the composite filter below can handle a single filter in the list.

Reply:
No, the composite filter fails :( This catch is for the specific 0/1 case.

Reply:
I'll add a clearer comment!